### ISSUE TITLE
New Rule: Prefer `Promise.resolve(foo)` to `new Promise((resolve) => resolve(foo))`

### ISSUE THREAD

**Issue by NotWoods** [eslint/eslint#17944]
### Rule details

Recommend using Promise static resolve/reject methods over `new Promise`

### Related ECMAScript feature

Promises

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

```js
new Promise((resolve) => resolve(foo));
new Promise((resolve) => { resolve(foo) });
// autofix to Promise.resolve(foo);

new Promise((_, reject) => reject(foo));
new Promise((_, reject) => { reject(foo) });
// autofix to Promise.reject(foo);
```

### Why should this rule be in the core instead of a plugin?

This rule relates to a core ESMAScript feature and fixes a common mistake when using promises

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

Using `new Promise` for simple use cases is a common mistake when developers first start working with promises. This rule would check for simple use cases, where resolve or reject is immediately called in the promise executor. If the executor function is more complex then the rule isn't triggered.

Possible name: prefer-promise-static-methods

**Comment by Rec0iL99** [comment]
Hi @NotWoods, thanks for the issue. This feels more like a stylistic preference. Plus, we currently only accept new rule proposals that relate to new ECMAScript features that reached stage 4 in the preceding 12 months ([rule policy](https://eslint.org/blog/2020/05/changes-to-rules-policies/#what-are-the-changes%3F)). This feels more like a rule for a plugin like [eslint-plugin-promise](https://github.com/eslint-community/eslint-plugin-promise).
